src/esp/scene/SceneNode.h

```diff
@@ -32,7 +32,8 @@ enum class SceneNodeType {
   OBJECT = 4,  // objects added via physics api
 };
 
-class SceneNode : public MagnumObject {
+class SceneNode : public MagnumObject,
+                  public Magnum::SceneGraph::AbstractFeature3D {
```

Comment:
My biggest concern comes from this inheritance. SceneNode is just a pure scene graph node, inherited from the MagnumObject, and a "feature" is an attachment on that node.
Logically, a class should not be a node and an attachment to that node at the same time. This is not a good design in my opinion.

Can we do it using "composition method" instead of "inheritance method"? Namely you created a new feature inherited from AbstractFeature3D, and add a "feature pointer" to this new feature in the SceneNode class?

Reply:
This is the way it was recommended by the magnum docs and I believe it is the only way to do it: https://doc.magnum.graphics/magnum/scenegraph.html#scenegraph-features-caching

Reply:
> Logically, a class should not be a node and an attachment to that node at the same time. 

That's supported *by design*, and perfectly fine. (Unless your coding guidelines ban multiple inheritance.)

Could be also a member variable, but then everything becomes more verbose, and you'll need another derived class and...